client/asset/dcr/rpcwallet.go

```diff
@@ -190,6 +191,72 @@ func newRPCWallet(settings map[string]string, logger dex.Logger, net dex.Network
 	return rpcw, nil
 }
 
+// Reconfigure updates the wallet to user a new configuration.
+func (w *rpcWallet) Reconfigure(ctx context.Context, cfg *asset.WalletConfig, net dex.Network, currentAddress, depositAccount string) (restart bool, err error) {
+	if !(cfg.Type == walletTypeDcrwRPC || cfg.Type == walletTypeLegacy) {
+		return true, nil
+	}
+
+	rpcCfg, chainParams, err := loadRPCConfig(cfg.Settings, net)
+	if err != nil {
+		return false, fmt.Errorf("error parsing config: %w", err)
+	}
+
+	walletCfg := new(walletConfig)
+	_, err = loadConfig(cfg.Settings, net, walletCfg)
+	if err != nil {
+		return false, err
+	}
+
+	if chainParams.Net != w.chainParams.Net {
+		return false, errors.New("cannot reconfigure to use different network")
+	}
+	certs, err := os.ReadFile(rpcCfg.RPCCert)
+	if err != nil {
+		return false, fmt.Errorf("TLS certificate read error: %w", err)
+	}
+	if rpcCfg.RPCUser == w.rpcCfg.User &&
+		rpcCfg.RPCPass == w.rpcCfg.Pass &&
+		bytes.Equal(certs, w.rpcCfg.Certificates) &&
+		rpcCfg.RPCListen == w.rpcCfg.Host {
+		return false, nil
+	}
+
+	newWallet, err := newRPCWallet(cfg.Settings, w.log, net)
+	if err != nil {
+		return false, err
+	}
+
+	err = newWallet.Connect(ctx)
+	if err != nil {
+		return false, fmt.Errorf("error connecting new wallet")
+	}
+
+	if walletCfg.ActivelyUsed {
+		a, err := stdaddr.DecodeAddress(currentAddress, w.chainParams)
+		if err != nil {
+			return false, err
+		}
+		owns, err := newWallet.AccountOwnsAddress(ctx, a, depositAccount)
+		if err != nil {
+			return false, err
+		}
+		if !owns {
+			return false, errors.New("cannot reconfigure to different wallet while there are active deals")
```

Comment:
I know what you're saying, but "deals" doesn't jive with all the rest of our messages

Reply:
changed it to "trades".